tools/explorer/pkg/workloads/reservation.go

```diff
@@ -372,6 +378,15 @@ func (a *API) workloads(r *http.Request) (interface{}, mw.Response) {
 			}
 		}
 
+		// if escrow is not enabled, we skip the pay state go to deploy directly
```

Comment:
Why is there specific logic being added for the `Free` escrow (and why here)? I did noticed that the `Free` escrow has only blanket implementations for the escrow interface - I think the logic to move from `Pay` to `Deploy` state should be moved to the `RegisterReservation` method. Then we have identical flow for either the rea or "fake" escrow (from the reservation standpoint ofc).